tests/SimplifyTest.elm

```diff
@@ -18256,6 +18257,156 @@ a = Sub.none
 
 
 
+-- Task
+
+
+taskTests : Test
+taskTests =
+    Test.describe "Task"
+        [ taskAndThenTests
+        ]
+
+
+taskAndThenTests : Test
+taskAndThenTests =
+    describe "Task.andThen"
+        [ test "should not report Task.andThen used with okay arguments" <|
+            \() ->
+                """module A exposing (..)
+a = Task.andThen f x
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectNoErrors
+        , test "should replace Task.andThen f (Task.fail z) by (Task.fail z)" <|
+            \() ->
+                """module A exposing (..)
+a = Task.andThen f (Task.fail z)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Task.andThen on a failing task will result in the given failing task"
+                            , details = [ "You can replace this call by the given failing task." ]
+                            , under = "Task.andThen"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = (Task.fail z)
```

Comment:
Could this lead to compiler errors because the type annotation is wrong? I don't think it could, but double-checking.

Reply:
andThen can't change the error type and `fail` produced a free variable for the success value (just like with result). So we should be good!